File: app/src/main/java/com/example/calendarbyourselvesdacs3/di/CalendarModule.java
package com.example.calendarbyourselvesdacs3.di;

import dagger.Module;
import dagger.Provides;
import dagger.hilt.InstallIn;
import dagger.hilt.components.SingletonComponent;

@Module
@InstallIn(SingletonComponent::class)
class CalendarModule {

    @Provides
    fun getMonthDays() = GetMonthDays()
}

File: app/src/main/java/com/example/calendarbyourselvesdacs3/di/CalendarModule.kt
package com.example.calendarbyourselvesdacs3.di

import com.example.calendarbyourselvesdacs3.data.repository.event.EventRepository
import com.example.calendarbyourselvesdacs3.domain.model.calendar.GetMonthDays
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import javax.inject.Singleton


@Module
@InstallIn(SingletonComponent::class)
class CalendarModule {

    @Provides
    fun getMonthDays() = GetMonthDays()

    @Provides
    @Singleton
    fun provideEventRepository(): EventRepository {
        return EventRepository()
    }



//    @Provides
//    @Singleton
//    fun provideSignInRepository(@ApplicationContext context: Context): GoogleAuthUiClient {
//        return GoogleAuthUiClient(
//            context = context,
//            oneTapClient = Identity.getSignInClient(context)
//        )
//    }
}